### PR TITLE
[DAT-18787] snapshot logic for tblProperties and table Location

### PR DESCRIPTION
**Pull request by KushnirykOleh**

### CODE REVIEW

File: src/main/java/liquibase/ext/databricks/change/createTable/ExtendedTableProperties.java [hunk 1]
```diff
@@ -15,20 +23,4 @@ public String getSerializedObjectName() {
     public String getSerializedObjectNamespace() {
         return "http://www.liquibase.org/xml/ns/databricks";
```
Comment (filipelautert) [inline src/main/java/liquibase/ext/databricks/change/createTable/ExtendedTableProperties.java:24]: As this class is being refactored, here we could also return  `NamespaceDetailsDatabricks.DATABRICKS_NAMESPACE`

File: src/main/java/liquibase/ext/databricks/snapshot/jvm/TableSnapshotGeneratorDatabricks.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+            if (tableProperty.get("COL_NAME").equals(DETAILED_TABLE_INFORMATION_NODE)) {
+                detailedInformationNode = true;
+                continue;
+            }
+            if (detailedInformationNode && tableProperty.get("COL_NAME").equals(LOCATION)) {
+                table.setAttribute(LOCATION, tableProperty.get("DATA_TYPE"));
+            }
+            if (detailedInformationNode && tableProperty.get("COL_NAME").equals(TABLE_PROPERTIES)) {
+                String tblProperties = (String) tableProperty.get("DATA_TYPE");
+                table.setAttribute(TBL_PROPERTIES, tblProperties.substring(1, tblProperties.length() - 1));// remove starting and ending square brackets
```
Comment (filipelautert) [inline src/main/java/liquibase/ext/databricks/snapshot/jvm/TableSnapshotGeneratorDatabricks.java:53]: Would make sense to add a `break` here to leave the for loop as we already found what we want?

**Review by filipelautert — CHANGES_REQUESTED** [review]

File: src/main/java/liquibase/ext/databricks/snapshot/jvm/TableSnapshotGeneratorDatabricks.java (hunk 2 above)
Comment (KushnirykOleh) [reply src/main/java/liquibase/ext/databricks/snapshot/jvm/TableSnapshotGeneratorDatabricks.java:53]: I'm unsure if we can rely on the order of `Location` and `tblProperties` rows to break in second block. I can create a var that will be incremented each time we find needed object and break if all are found
Something like
```
        byte nodesFound = 0;
        for (Map<String, ?> tableProperty : tablePropertiesResponse) {
            if (tableProperty.get("COL_NAME").equals(DETAILED_TABLE_INFORMATION_NODE)) {
                detailedInformationNode = true;
                continue;
            }
            if (detailedInformationNode && tableProperty.get("COL_NAME").equals(LOCATION)) {
                table.setAttribute(LOCATION, tableProperty.get("DATA_TYPE"));
                nodesFound++;
            }
            if (detailedInformationNode && tableProperty.get("COL_NAME").equals(TABLE_PROPERTIES)) {
                String tblProperties = (String) tableProperty.get("DATA_TYPE");
                table.setAttribute(TBL_PROPERTIES, tblProperties.substring(1, tblProperties.length() - 1));// remove starting and ending square brackets
                nodesFound++;
            }
            if (nodesFound == 2) {
                break;
            }
        }
        ```

Comment (filipelautert) [reply src/main/java/liquibase/ext/databricks/snapshot/jvm/TableSnapshotGeneratorDatabricks.java:53]: Oh I see. So no worries, makes sense now!

**Review by filipelautert — APPROVED** [review]

**Pull request merged**